src/common/memory.rs

```diff
@@ -21,6 +21,34 @@ struct MemoryMapEntry {
     acpi: u32,
 }
 
+struct Memory {
+    address: usize,
+}
+
+impl Memory {
+    pub fn new(size: usize) -> Option<Self> {
+        let alloc = alloc(size);
+        if alloc > 0 {
+            Some(Memory { address: alloc })
+        }
+        else { None }
+    }
+
+    pub fn renew(&self, size: usize) -> Option<Self> {
+        let realloc = realloc(self.address, size);
```

Comment:
oh wait, nevermind. I'm stupid. What you suggest did work after trying it again.